File: examples/python/mediafire.py
import requests
import os

API_KEY = os.environ.get('COGNIMA_API_KEY', 'ck_your_api_key')
BASE_URL = 'https://cog2.cognima.com.br/api/v1'

def get_mediafire_info():
    """Obter informações de arquivo do MediaFire"""
    try:
        url = 'https://www.mediafire.com/file/abc123xyz/arquivo.zip/file'
        
        response = requests.get(
            f'{BASE_URL}/mediafire/info',
            params={'url': url},
            headers={'Authorization': f'Bearer {API_KEY}'}
        )
        response.raise_for_status()
        
        data = response.json()['data']
        print('📁 MediaFire File Info:')
        print(f"   Nome: {data['fileName']}")
        print(f"   Tamanho: {data['fileSize']}")
        print(f"   Data Upload: {data['uploadDate']}")
        print(f"   Tipo: {data['mimetype']}")
        print(f"   Extensão: {data['extension']}")
        print(f"   Link: {data['downloadUrl']}")
        
        return data
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')
        if hasattr(e, 'response') and e.response:
            print(f'   Detalhes: {e.response.json()}')

def download_mediafire():
    """Obter link de download do MediaFire"""
    try:
        url = 'https://www.mediafire.com/file/abc123xyz/arquivo.zip/file'
        
        response = requests.get(
            f'{BASE_URL}/mediafire/download',
            params={'url': url},
            headers={'Authorization': f'Bearer {API_KEY}'}
        )
        response.raise_for_status()
        
        data = response.json()['data']
        print('✅ Download disponível:')
        print(f"   Arquivo: {data['fileName']}")
        print(f"   URL: {data['downloadUrl']}")
        
        return data
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')

def download_file_to_disk(mediafire_url: str, output_path: str):
    """Baixar arquivo diretamente para o disco"""
    try:
        # Primeiro, obter o link de download
        response = requests.get(
            f'{BASE_URL}/mediafire/download',
            params={'url': mediafire_url},
            headers={'Authorization': f'Bearer {API_KEY}'}
        )
        response.raise_for_status()
        
        download_url = response.json()['data']['downloadUrl']
        file_name = response.json()['data']['fileName']
        
        # Depois, baixar o arquivo
        print(f'⬇️  Baixando {file_name}...')
        file_response = requests.get(download_url, stream=True)
        file_response.raise_for_status()
        
        with open(output_path, 'wb') as f:
            for chunk in file_response.iter_content(chunk_size=8192):
                f.write(chunk)
        
        print(f'✅ Arquivo salvo em: {output_path}')
        
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')

if __name__ == '__main__':
    print('=== MediaFire API Examples ===\n')
    get_mediafire_info()

File: examples/nodejs/mediafire.js
const axios = require('axios');

const API_KEY = process.env.COGNIMA_API_KEY || 'ck_your_api_key';
const BASE_URL = 'https://cog.api.br/api/v1';

/**
 * Obter informações de arquivo do MediaFire
 */
async function getMediaFireInfo() {
  try {
    const url = 'https://www.mediafire.com/file/abc123xyz/arquivo.zip/file';
    
    const response = await axios.get(
      `${BASE_URL}/mediafire/info`,
      {
        params: { url },
        headers: {
          'Authorization': `Bearer ${API_KEY}`
        }
      }
    );
    
    const data = response.data.data;
    console.log('📁 MediaFire File Info:');
    console.log(`   Nome: ${data.fileName}`);
    console.log(`   Tamanho: ${data.fileSize}`);
    console.log(`   Data Upload: ${data.uploadDate}`);
    console.log(`   Tipo: ${data.mimetype}`);
    console.log(`   Extensão: ${data.extension}`);
    console.log(`   Link: ${data.downloadUrl}`);
    
    return data;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Obter link de download do MediaFire
 */
async function downloadMediaFire() {
  try {
    const url = 'https://www.mediafire.com/file/abc123xyz/arquivo.zip/file';
    
    const response = await axios.get(
      `${BASE_URL}/mediafire/download`,
      {
        params: { url },
        headers: {
          'Authorization': `Bearer ${API_KEY}`
        }
      }
    );
    
    const data = response.data.data;
    console.log('✅ Download disponível:');
    console.log(`   Arquivo: ${data.fileName}`);
    console.log(`   URL: ${data.downloadUrl}`);
    
    // Para redirecionar diretamente, use: params: { url, redirect: 'true' }
    
    return data;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Download com redirecionamento automático
 */
async function downloadWithRedirect() {
  try {
    const url = 'https://www.mediafire.com/file/abc123xyz/arquivo.zip/file';
    
    // Com redirect=true, a API redireciona direto para o arquivo
    const downloadUrl = `${BASE_URL}/mediafire/download?url=${encodeURIComponent(url)}&redirect=true`;
    
    console.log('🔗 URL para download direto:');
    console.log(`   ${downloadUrl}`);
    console.log('   (Adicione o header Authorization para usar)');
    
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

// Executar exemplos
console.log('=== MediaFire API Examples ===\n');
getMediaFireInfo();

